test/system/irohad_test.cpp

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2018 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+#include <rapidjson/document.h>
+#include <boost/filesystem.hpp>
+#include <boost/process.hpp>
+#include <pqxx/pqxx>
+
+#include "common/files.hpp"
+#include "main/iroha_conf_loader.hpp"
+
+using namespace boost::process;
+using namespace std::chrono_literals;
+
+class IrohadTest : public testing::Test {
+ public:
+  virtual void SetUp() {
+    setPaths();
+    auto config = parse_iroha_config(path_config.string());
+    blockstore_path = config[config_members::BlockStorePath].GetString();
+    pgopts = config[config_members::PgOpt].GetString();
+  }
+  virtual void TearDown() {
+    iroha::remove_all(blockstore_path);
+    dropPostgres();
+  }
+
+ private:
+  void setPaths() {
+    path_irohad = boost::filesystem::path(PATHIROHAD);
+    irohad_executable = path_irohad / "irohad";
+    path_example = path_irohad / ".." / ".." / "example";
+    path_config = path_example / "config.sample";
+    path_genesis = path_example / "genesis.block";
+    path_keypair = path_example / "node0";
+    params = " --config " + path_config.string() + " --genesis_block "
+        + path_genesis.string() + " --keypair_name " + path_keypair.string();
+    timeout = 1s;
+  }
+
+  void dropPostgres() {
+    connection = std::make_shared<pqxx::lazyconnection>(pgopts);
+    try {
+      connection->activate();
+    } catch (const pqxx::broken_connection &e) {
+      FAIL() << "Connection to PostgreSQL broken: " << e.what();
```

Comment:
This may fail the tests if there's a problem with psql connection. And actually there's may be no such one. Imo it's better print some warning and that's enough

Reply:
Test will be failed anyway because irohad uses the same connection with the same parameters.